internal/ui/restore/progressformatter/progressformatter.go

```diff
@@ -0,0 +1,119 @@
+package progressformatter
```

Comment:
Please remove the intermediate `progressformatter` package and place the file directly into `internal/ui/restore`. To import a package with a different name you could use something similar to  `import rtest "github.com/restic/restic/internal/test"`. Just add the necessary interface directly to `internal/restore`.

Reply:
So something like

    import restoreui "github.com/restic/restic/internal/ui/restore"

If you also rename the RestoreProgressFormatter to ProgressFormatter (or even just Progress), the name is less repetitive: restoreui.Progress(Formatter).